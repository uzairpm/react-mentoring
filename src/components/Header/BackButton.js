import React, {Component} from 'react';
import { Link } from "react-router-dom";

const BackButton = (props) => (
    <Link to="/" className="btn btn-primary btn-sm widerButton">SEARCH</Link>
);
export default BackButton;
