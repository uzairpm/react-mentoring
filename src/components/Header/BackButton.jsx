import React from 'react';
import { Link } from "react-router-dom";
import styled from 'styled-components';

const WiderLink = styled(Link)`
    padding: 0.25rem 2rem;
`;

const BackButton = (props) => (
    <WiderLink to="/" className="btn btn-primary btn-sm">SEARCH</WiderLink>
);
export default BackButton;
