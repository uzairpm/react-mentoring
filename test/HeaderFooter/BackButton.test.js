import React from 'react';
import { Link } from "react-router-dom";

import BackButton from '../../src/components/Header/BackButton';

describe('Check Component creation', () => {
    test('BackButton component', () => {
        const element = shallow(<BackButton />);
        expect(element).toMatchSnapshot();
    });
});