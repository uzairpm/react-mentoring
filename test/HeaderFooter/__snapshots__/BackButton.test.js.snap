// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Component creation BackButton component snapshot test 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BackButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "SEARCH",
      "className": "btn btn-primary btn-sm widerButton",
      "replace": false,
      "to": "/",
    },
    "ref": null,
    "rendered": "SEARCH",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "SEARCH",
        "className": "btn btn-primary btn-sm widerButton",
        "replace": false,
        "to": "/",
      },
      "ref": null,
      "rendered": "SEARCH",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
